/*
*  __________________________________________________________________________
* /                                                                          \
* |                           _                                              |
* |                          / \                         _                   |
* |                  ___    / _ \   _ __ ___   ____  ___| |                  |
* |                 / _ \  / /_\ \ | '_ ` _ \./  _ \/  _  |                  |
* |                |  __/ /  ___  \| | | | | |  (_) ) (_) |                  |
* |                 \___|/__/   \__\_| |_| |_|\____/\_____/                  |
* |                                                                          |
* |                              eAmod Script                                |
* |                                                                          |
* |--------------------------------------------------------------------------|
* | Script Name: No Storage Mapflag                                          |
* |--------------------------------------------------------------------------|
* | Created by: shellTeMP                                                    |
* |--------------------------------------------------------------------------|
* | Versions eASU/eA/brA/Herc/rA: all/all                                    |
* |--------------------------------------------------------------------------|
* | Description: Block the storage.                                          |
* \_________________________________________________________________________/
*/
Index: src/map/atcommand.c
===================================================================
--- src/map/atcommand.c	(revision 1187)
+++ src/map/atcommand.c	(working copy)
@@ -818,6 +818,12 @@
 
 	if (sd->npc_id || sd->state.vending || sd->state.buyingstore || sd->state.trading || sd->state.storage_flag)
 		return -1;
+		
+	if(map[sd->bl.m].nostorage == 1 || map[sd->bl.m].nostorage > 2)
+	{
+		clif_displaymessage(sd->fd,"O uso do armazém não é permitido nesse mapa.");
+		return -1;
+	}
 
 	if (storage_storageopen(sd) == 1)
 	{	//Already open.
@@ -845,6 +851,12 @@
 
 	if (sd->npc_id || sd->state.vending || sd->state.buyingstore || sd->state.trading)
 		return -1;
+		
+	if(map[sd->bl.m].nostorage > 1)
+	{
+		clif_displaymessage(sd->fd,"O uso do armazém não é permitido nesse mapa.");
+		return -1;
+	}
 
 	if (sd->state.storage_flag == 1) {
 		clif_displaymessage(fd, msg_txt(250));
Index: src/map/map.c
===================================================================
--- src/map/map.c	(revision 1187)
+++ src/map/map.c	(working copy)
@@ -2940,6 +2940,7 @@
 		map[i].nocommand = 0;  // nocommand mapflag level
 		map[i].bexp      = 100;  // per map base exp multiplicator
 		map[i].jexp      = 100;  // per map job exp multiplicator
+		map[i].nostorage = 0; // [Shiraz]
 		memset(map[i].drop_list, 0, sizeof(map[i].drop_list));  // pvp nightmare drop list
 
 		// adjustments
Index: src/map/map.h
===================================================================
--- src/map/map.h	(revision 1187)
+++ src/map/map.h	(working copy)
@@ -559,6 +559,7 @@
 	int jexp;	// map experience multiplicator
 	int bexp;	// map experience multiplicator
 	int nocommand; //Blocks @/# commands for non-gms. [Skotlex]
+	int nostorage; // [Shiraz]
 	/**
 	 * Ice wall reference counter for bugreport:3574
 	 * - since there are a thounsand mobs out there in a lot of maps checking on,
Index: src/map/npc.c
===================================================================
--- src/map/npc.c	(revision 1187)
+++ src/map/npc.c	(working copy)
@@ -3237,6 +3237,8 @@
 		map[m].flag.guildlock=state;
 	else if (!strcmpi(w3,"reset"))
 		map[m].flag.reset=state;
+	else if(!strcmpi(w3,"nostorage"))
+		map[m].nostorage = atoi(w4);
 	else
 		ShowError("npc_parse_mapflag: mapflag nao reconhecido '%s' (arquivo '%s', linha '%d').\n", w3, filepath, strline(buffer,start-buffer));
 
Index: src/map/storage.c
===================================================================
--- src/map/storage.c	(revision 1187)
+++ src/map/storage.c	(working copy)
@@ -102,6 +102,12 @@
 		clif_displaymessage(sd->fd, msg_txt(246));
 		return 1;
 	}
+	
+	if(map[sd->bl.m].nostorage == 1 || map[sd->bl.m].nostorage > 2)
+	{
+		clif_displaymessage(sd->fd,"O uso do armazém não é permitido nesse mapa.");
+		return 1;
+	}
 
 	sd->state.storage_flag = 1;
 	storage_sortitem(sd->status.storage.items, ARRAYLENGTH(sd->status.storage.items));
@@ -365,6 +371,12 @@
 		clif_displaymessage(sd->fd, msg_txt(246));
 		return 1;
 	}
+	
+	if(map[sd->bl.m].nostorage > 1)
+	{
+		clif_displaymessage(sd->fd,"O uso do armazém não é permitido nesse mapa.");
+		return 2;
+	}
 
 	if((gstor = guild2storage2(sd->status.guild_id)) == NULL) {
 		intif_request_guild_storage(sd->status.account_id,sd->status.guild_id);
